dev/release/README.md

```diff
@@ -17,29 +17,34 @@ specific language governing permissions and limitations
 under the License.
 -->
 
-# Comet Release Process
+# Aapche DataFusion Comet: Source Release Process
 
 This documentation is for creating an official source release of Apache DataFusion Comet.
 
-The release process is based on the parent Apache DataFusion project, so please refer to the
-[DataFusion Release Process](https://github.com/apache/datafusion/blob/main/dev/release/README.md) for detailed
-instructions if you are not familiar with the release process here.
-
-Here is a brief overview of the steps involved in creating a release:
-
 ## Creating the Release Candidate
 
 This part of the process can be performed by any committer.
 
-- Create and merge a PR to update the version number & update the changelog
-- Push a release candidate tag (e.g. 0.1.0-rc1) to the Apache repository
+Here are the steps, using the 0.1.0 release as an example.
+
+### Create Release Branch
+
+Create a release branch from the latest commit in main and push to the Apache repo:
+
+```shell
+get fetch apache
```

Comment:
`apache` is a different repo?

Reply:
That's the name I use for my remote, but it would be good to clarify this in the docs.

```
$ git remote -v | grep apach
apache	git@github.com:apache/datafusion-comet.git (fetch)
apache	git@github.com:apache/datafusion-comet.git (push)
```

Reply:
Yea, I thought it might be a nickname for a particular repo. It'd be good to make it clear in the doc.